# Imports

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Data Description


In [3]:
data = pd.read_csv("market_prices.csv")
data.head(10) #prinnt first to rows

Unnamed: 0  adm0_id adm0_name  adm1_id      adm1_name  mkt_id mkt_name  \
0      135405       94     Ghana     1328  Greater Accra     209    Accra   
1      135406       94     Ghana     1328  Greater Accra     209    Accra   
2      135407       94     Ghana     1328  Greater Accra     209    Accra   
3      135408       94     Ghana     1328  Greater Accra     209    Accra   
4      135409       94     Ghana     1328  Greater Accra     209    Accra   
5      135410       94     Ghana     1328  Greater Accra     209    Accra   
6      135411       94     Ghana     1328  Greater Accra     209    Accra   
7      135412       94     Ghana     1328  Greater Accra     209    Accra   
8      135413       94     Ghana     1328  Greater Accra     209    Accra   
9      135414       94     Ghana     1328  Greater Accra     209    Accra   

   cm_id cm_name  cur_id cur_name  pt_id    pt_name  um_id um_name  mp_month  \
0     51   Maize      35      GHS     14  Wholesale      9  100 KG         1   
1     51   Maize      35      GHS     14  Wholesale      9  100 KG         2   
2     51   Maize      35      GHS     14  Wholesale      9  100 KG         3   
3     51   Maize      35      GHS     14  Wholesale      9  100 KG         4   
4     51   Maize      35      GHS     14  Wholesale      9  100 KG         5   
5     51   Maize      35      GHS     14  Wholesale      9  100 KG         6   
6     51   Maize      35      GHS     14  Wholesale      9  100 KG         7   
7     51   Maize      35      GHS     14  Wholesale      9  100 KG         8   
8     51   Maize      35      GHS     14  Wholesale      9  100 KG         9   
9     51   Maize      35      GHS     14  Wholesale      9  100 KG        10   

   mp_year  mp_price mp_commoditysource  
0     2008     30.75               MOFA  
1     2008     34.25               MOFA  
2     2008     41.67               MOFA  
3     2008     46.50               MOFA  
4     2008     52.00               MOFA  
5     2008     62.25               MOFA  
6     2008     81.60               MOFA  
7     2008     59.75               MOFA  
8     2008     43.50               MOFA  
9     2008     46.50               MOFA

In [4]:
# Distribution view of the data
data.describe()

Unnamed: 0  adm0_id      adm1_id       mkt_id        cm_id  cur_id  \
count    8112.000000   8112.0  8112.000000  8112.000000  8112.000000  8112.0   
mean   139460.500000     94.0  1327.331977   215.919872    76.564965    35.0   
std      2341.877025      0.0     2.882634     4.239035    18.492255     0.0   
min    135405.000000     94.0  1324.000000   209.000000    51.000000    35.0   
25%    137432.750000     94.0  1325.000000   212.000000    64.000000    35.0   
50%    139460.500000     94.0  1327.000000   216.000000    71.000000    35.0   
75%    141488.250000     94.0  1329.000000   220.000000    98.000000    35.0   
max    143516.000000     94.0  1333.000000   223.000000    99.000000    35.0   

        pt_id        um_id     mp_month      mp_year     mp_price  
count  8112.0  8112.000000  8112.000000  8112.000000  8112.000000  
mean     14.0    19.041297     6.225222  2012.325197   105.432938  
std       0.0     5.084166     3.443353     2.720356   117.479997  
min      14.0     9.000000     1.000000  2008.000000     0.870000  
25%      14.0    20.000000     3.000000  2010.000000    26.250000  
50%      14.0    21.000000     6.000000  2012.000000    66.250000  
75%      14.0    22.000000     9.000000  2015.000000   144.000000  
max      14.0    23.000000    12.000000  2017.000000  1000.000000

In [8]:
# general information about the data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8112 entries, 0 to 8111
Data columns (total 19 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Unnamed: 0          8112 non-null   int64  
 1   adm0_id             8112 non-null   int64  
 2   adm0_name           8112 non-null   object 
 3   adm1_id             8112 non-null   int64  
 4   adm1_name           8112 non-null   object 
 5   mkt_id              8112 non-null   int64  
 6   mkt_name            8112 non-null   object 
 7   cm_id               8112 non-null   int64  
 8   cm_name             8112 non-null   object 
 9   cur_id              8112 non-null   int64  
 10  cur_name            8112 non-null   object 
 11  pt_id               8112 non-null   int64  
 12  pt_name             8112 non-null   object 
 13  um_id               8112 non-null   int64  
 14  um_name             8112 non-null   object 
 15  mp_month            8112 non-null   int64  
 16  mp_yea

In [9]:
# Checking for null
data.isnull().sum()

Unnamed: 0            0
adm0_id               0
adm0_name             0
adm1_id               0
adm1_name             0
mkt_id                0
mkt_name              0
cm_id                 0
cm_name               0
cur_id                0
cur_name              0
pt_id                 0
pt_name               0
um_id                 0
um_name               0
mp_month              0
mp_year               0
mp_price              0
mp_commoditysource    0
dtype: int64

## Summary initial observation of the data

* The data is occupying a space of 1.2 MB
* The data contains 8112 rows and 19 columns. One of the columns is Unnamed.
* One column is of float64 datatype, ten of int64 datatype and eight are string or object datatype
* There are no missing values